### PR TITLE
*: clean up the logging component

### PR DESCRIPTION
**Pull request by SabaPing**
### What problem does this PR solve?

> Please refer to the meta issue for more information. #24190

The primary goal is to clean up the logging component. TiDB depends on two different third part log components, which are logrus and zap. This creates unnecessary complexity. In addition, there are some other logging flaws in code.

### What is changed and how it works?

What's Changed: 

- Migrate logrus dependency to `pingcap/log`, which is based on zap.
- Remove invalid log format `console`.
- Cleanup gRPC debug log.

**Outside users should not perceive relevant changes.**

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note.

### CODE REVIEW

**Comment by SabaPing** [comment]
This is the first step for the task "Unify Log Framework".
After this pr being merged, I'll cleanup log code in pd, br and tidb-dashboard, one-by-one.
FYI, please read dev design here: [here](https://pingcap.feishu.cn/docs/doccnlXdhYAcl4JeBnu3qFNXctc)

**Comment by zz-jason** [comment]
@SabaPing thanks for your contribution, could you file a PR to discuss the proposal in the community?

**Comment by SabaPing** [comment]
> @SabaPing thanks for your contribution, could you file a PR to discuss the proposal in the community?
> 
> Edit (2021/03/29): you can find the proposal template here: https://github.com/pingcap/tidb/blob/master/docs/design/TEMPLATE.md

Sure I'll finish the proposal in this week.

**Comment by breeswish** [comment]
> > @SabaPing thanks for your contribution, could you file a PR to discuss the proposal in the community?
> > Edit (2021/03/29): you can find the proposal template here: https://github.com/pingcap/tidb/blob/master/docs/design/TEMPLATE.md
> 
> Sure I'll finish the proposal in this week.

I think you can just post your doc in the sig channel. However remember to make it public. @SabaPing

**Comment by crazycs520** [comment]
@AilinKid @tangenta PTAL

File: util/logutil/slow_query_logger.go
Comment (crazycs520) [inline util/logutil/slow_query_logger.go:46]:
```suggestion
	fmt.Fprintf(b, "# Time: %s\n", entry.Time.Format(SlowLogTimeFormat))
```

File: go.mod
Comment (crazycs520) [inline go.mod:62]: Can we remove it now?

Comment (xhebox) [reply go.mod:62]: Indirect can not be removed. It is added by go tools.

Comment (crazycs520) [reply go.mod:62]: In my local environment, I can remove this line by manual, then run `make tidy` will delete the related line in `go.sum`.

Comment (SabaPing) [reply go.mod:62]: I'll keep `indirect` keyword here.

**Review by crazycs520 — COMMENTED** [review]
/LGTM

**Review by xhebox — APPROVED** [review]
LGTM

File: go.mod
Comment (dragonly) [reply go.mod:62]: per `go help modules`:
```
As part of maintaining the require statements in go.mod, the go command
tracks which ones provide packages imported directly by the current module
and which ones provide packages only used indirectly by other module
dependencies. Requirements needed only for indirect uses are marked with a
"// indirect" comment in the go.mod file. Indirect requirements are
automatically removed from the go.mod file once they are implied by other
direct requirements. Indirect requirements only arise when using modules
that fail to state some of their own dependencies or when explicitly
upgrading a module's dependencies ahead of its own stated requirements.
```

**Comment by zz-jason** [comment]
@SabaPing any update about the proposal PR?

**Comment by SabaPing** [comment]
> @SabaPing any update about the proposal PR?

I've added a simple proposal in the latest commit.

**Comment by xhebox** [comment]
@zz-jason @SabaPing I think the acceptation of RFC is more formal, but this PR is a little bit special. Do we need to notify everyone the changes of logging? Can we just merge the PR with the RFC, or how do we do?

**Comment by tisonkun** [comment]
> @zz-jason @SabaPing I think the acceptation of RFC is more formal, but this PR is a little bit special. Do we need to notify everyone the changes of logging? Can we just merge the PR with the RFC, or how do we do?

@xhebox @SabaPing Is logging a user facing interface? I'm unsure whether in go world user can config logging property as in java world using slf4j / log4j. If so, we'd better create a separated proposal to mention all changes of user facing interface. And hold this pull request until the proposal accepted and merged. Hopefully we will catch the progress in time so that we don't have one more stale pr.

But if logging in go world is internal or standard, in other words, user can do litter on configure logging property or the approach to configure logging property between zap & logrus is same. Then I think it is ok for driving this effort without a proposal.

**Comment by tisonkun** [comment]
... and I found this effort consists of changes among several projects, you'd better create a proposal for the whole story and have a tracking issue for all this subtasks.

**Comment by dragonly** [comment]
@tisonkun What do you mean by "several projects"?
And there's a [design doc](https://github.com/pingcap/tidb/pull/23534/files#diff-98e78d710b6e8359ce28a2069ba9c835e98fd11726126e7a73a72da1879aed01) in this PR.

**Comment by tisonkun** [comment]
> @tisonkun What do you mean by "several projects"?
> And there's a [design doc](https://github.com/pingcap/tidb/pull/23534/files#diff-98e78d710b6e8359ce28a2069ba9c835e98fd11726126e7a73a72da1879aed01) in this PR.

The whole story of this "unify logging component" includes tidb / br / ...

And yes there is a design doc. I have a short offline discussion with @SabaPing and suggest we separate the proposal pr for consensus on the design firstly and quickly, later with detailed development tasks.

**Comment by xhebox** [comment]
@SabaPing Rabase plz? I think we can merge the PR now.

**Comment by SabaPing** [comment]
> @SabaPing Rabase plz? I think we can merge the PR now.

Done rebased

File: go.mod
Comment (breeswish) [reply go.mod:57]: Yeah I think it would be better to remove indirect dependencies.

File: tidb-server/main.go
Comment (breeswish) [inline tidb-server/main.go:597]: The bug comes from previously we just Discard these logs @SabaPing

File: go.mod
Comment (tisonkun) [reply go.mod:57]:
```
$ go mod why github.com/sirupsen/logrus
# github.com/sirupsen/logrus
github.com/pingcap/tidb/ddl/util
github.com/pingcap/tidb/ddl/util.test
go.etcd.io/etcd/integration
go.etcd.io/etcd/embed
github.com/tmc/grpc-websocket-proxy/wsproxy
github.com/sirupsen/logrus
```

`go mod tidy` will add it back due to etcd dependency maybe.

File: util/logutil/log.go
Comment (tisonkun) [inline util/logutil/log.go:101]: PD still depend on this method, right?

Comment (tisonkun) [reply util/logutil/log.go:101]: If so, why not we retain `InitLogger` and drop `InitZapLogger`? An interface is better to be more abstract.

**Pull request merged**
